server/events.go

```diff
@@ -1495,6 +1495,12 @@ func (s *Server) userInfoReq(sub *subscription, c *client, _ *Account, subject,
 		return
 	}
 
+	// Only the server that the client is connected to should answer, otherwise
+	// we may not know about their connection JWT etc.
+	if ci.Server != s.Name() {
```

Comment:
We have ways for subs to not propagate interest to other servers, the sub for this one should have been that I believe. Will check the original sub code for that endpoint.

Reply:
Switched to this approach instead, seems to solve the problem more elegantly.